internal/idle/idle_e2e_test.go

```diff
@@ -234,16 +247,14 @@ func (s) TestChannelIdleness_Enabled_OngoingCall(t *testing.T) {
 			errCh <- err
 			return
 		}
-
-		// Unblock the unary RPC on the server.
 		errCh <- nil
 	}()
 
-	// Make a unary RPC that blocks on the server, thereby ensuring that the
-	// count of active RPCs on the client is non-zero.
+	// Start a streaming RPC. Even though the stream is inactive (i,e no reads
+	// or writes), the stream should prevent the channel from moving to IDLE.
 	client := testgrpc.NewTestServiceClient(cc)
-	if _, err := client.EmptyCall(ctx, &testpb.Empty{}); err != nil {
-		t.Errorf("EmptyCall RPC failed: %v", err)
+	if _, err := client.FullDuplexCall(ctx); err != nil {
```

Comment:
You should keep the existing test for unary too.

But unary and streaming should share code better in the actual code.

There are currently 2 copies of accounting for RPCs, but there should be only one.

If we do this in `newClientStream` instead of the two places it's done, that should take care of both.

Reply:
> You should keep the existing test for unary too.

Done.

> If we do this in newClientStream instead of the two places it's done, that should take care of both.

It would. But we would increment the call count only after the interceptors get to run and if the channel moves to IDLE when the interceptors are running, is that going to be OK?

Reply:
> It would. But we would increment the call count only after the interceptors get to run and if the channel moves to IDLE when the interceptors are running, is that going to be OK?

It's not ideal, but the window of that race is miniscule.  Like who waits 30 minutes to start an RPC but then 100ns would have saved them?  I'm not worried about this.

The other nice thing is that the interceptors can't see the call option you're adding, which means they can't (accidentally) remove it and prevent our accounting from working correctly.

Reply:
Done.

> The other nice thing is that the interceptors can't see the call option you're adding, which means they can't (accidentally) remove it and prevent our accounting from working correctly.

+1